tests/persistence/test_persistence_leverage.py

```diff
@@ -0,0 +1,638 @@
+        total_profit_percentage = total_profit / stake_value
+            = 0.00018628848193889054 / 0.0010024999999225066
+            = 0.18582392214792087
+    """
+    trade = Trade(
+        pair='ETH/BTC',
+        stake_amount=0.0009999999999226999,
+        open_rate=0.01,
+        amount=5,
+        open_date=datetime.utcnow() - timedelta(hours=4, minutes=55),
+        fee_open=fee.return_value,
+        fee_close=fee.return_value,
+        exchange='binance',
+        interest_rate=0.0005,
+        interest_mode=InterestMode.HOURSPERDAY
+    )
+    trade.open_order_id = 'something'
+    trade.update(limit_lev_buy_order)
+    assert trade._calc_open_trade_value() == 0.00300749999976752
+    trade.update(limit_lev_sell_order)
+
+    # Is slightly different due to compilation time changes. Interest depends on time
+    assert round(trade.calc_close_trade_value(), 11) == round(0.003193788481706411, 11)
+    # Profit in BTC
+    assert round(trade.calc_profit(), 8) == round(0.00018628848193889054, 8)
+    # Profit in percent
+    assert round(trade.calc_profit_ratio(), 8) == round(0.18582392214792087, 8)
+
+
+@pytest.mark.usefixtures("init_persistence")
+def test_trade_close_lev(fee):
+    """
+        5 hour leveraged market trade on Kraken at 3x leverage
+        fee: 0.25% base
+        interest_rate: 0.05% per 4 hrs
+        open_rate: 0.1 base
+        close_rate: 0.2 base
+        amount: 5 * leverage(3) = 15 crypto
+        stake_amount: 0.5
+        borrowed: 1 base
+        time-periods: 5/4 periods of 4hrs
+        interest: borrowed * interest_rate * ceil(1 + time-periods)
+                    = 1 * 0.0005 * ceil(9/4) = 0.0015 crypto
+        open_value: (amount * open_rate) + (amount * open_rate * fee)
+            = (15 * 0.1) + (15 * 0.1 * 0.0025)
+            = 1.50375
+        close_value: (amount * close_rate) + (amount * close_rate * fee) - interest
+            = (15 * 0.2) - (15 * 0.2 * 0.0025) - 0.0015
+            = 2.991
+        total_profit = close_value - open_value
+            = 2.991 - 1.50375
+            = 1.4872500000000002
+        total_profit_ratio = ((close_value/open_value) - 1) * leverage
+            = ((2.991/1.50375) - 1) * 3
+            = 2.96708229426434
+    """
+    trade = Trade(
+        pair='ETH/BTC',
+        stake_amount=0.5,
+        open_rate=0.1,
+        amount=15,
+        is_open=True,
+        fee_open=fee.return_value,
+        fee_close=fee.return_value,
+        open_date=datetime.utcnow() - timedelta(hours=4, minutes=55),
+        exchange='kraken',
+        leverage=3.0,
+        interest_rate=0.0005,
+        interest_mode=InterestMode.HOURSPER4
+    )
+    assert trade.close_profit is None
+    assert trade.close_date is None
+    assert trade.is_open is True
+    trade.close(0.2)
+    assert trade.is_open is False
+    assert trade.close_profit == round(2.96708229426434, 8)
+    assert trade.close_date is not None
+
+    # TODO-mg: Remove these comments probably
```

Comment:
yes?

Reply:
@samgermain forgot to push?

Reply:
Haven't pushed yet, I take it I should only click resolve once I push

Reply:
yes(ish) ... 
i guess it's fine if you've got it commited and it's "ready to push" - waiting for some further changes.

Otherwise, it'll become dangerous for both of us to forget ...